### PR TITLE
Feature/add styling to form

### PR DESCRIPTION
**Pull request by FriedSock**

### CODE REVIEW

File: app/views/defence_requests/_form.html.erb [hunk 1]
```diff
@@ -1,107 +1,185 @@
...
-      <label class="block-label" for="appropriate_adult">
-        <%= f.check_box :appropriate_adult %>
-        <%= t('appropriate_adult') %>
-      </label>
-    </fieldset>
-  </div>
+  <div id='main-form-body' <%= "class=hidden" if @defence_request.new_record?%>>
+    <!-- details -->
+    <div id='solicitor-details' <%= "class=hidden" if @defence_request.new_record? %>>
+      <h3><% t('solicitor_details') %></h3>
```
Comment (ansonK) [inline app/views/defence_requests/_form.html.erb]: this translation is missing

**Comment by ansonK** [comment]
LGTM
too much work needed to handle error states on input elements in a nice way atm

File: app/controllers/defence_requests_controller.rb [hunk 2]
```diff
@@ -133,22 +133,50 @@ def defence_request_params
...
     raw_params.tap do |p|
       email = p.delete("solicitor_email")
       solicitor = User.solicitors.find_by_email email
       p["solicitor_id"] = solicitor.id if solicitor
+
+      year = p.delete('date_of_birth_year')
+      month = p.delete('date_of_birth_month')
+      day = p.delete('date_of_birth_day')
+      if !(day.blank? || month.blank? && year.blank?)
+        p["date_of_birth"] = DateTime.new year.to_i, month.to_i, day.to_i
```
Comment (dp6ai) [inline app/controllers/defence_requests_controller.rb]: ??

File: app/views/defence_requests/_defence_requests_create_js.html.erb [hunk 3]
```diff
@@ -0,0 +1,71 @@
+<script>
```
Comment (dp6ai) [inline app/views/defence_requests/_defence_requests_create_js.html.erb]: think this is repeated elsewhere

File: app/views/defence_requests/_defence_requests_new_js.html.erb [hunk 4]
```diff
@@ -1,48 +1 @@
...
-    $(document).on('keydown', function (event) {
-        if ($('.solicitor_results_list *').length > 0 && event.keyCode === 27) {
-            $('.solicitor_results_list').empty();
-        }
-    });
-
-    var ownSolicitor_checked = ($("input[type='radio']#defence_request_solicitor_type_own:checked").length > 0);
-
-    var own_setup = function () {
-        $('#defence_request_solicitor_firm').removeAttr("disabled");
```
Comment (dp6ai) [inline app/views/defence_requests/_defence_requests_new_js.html.erb]: Weird javascript variable names...

File: spec/models/defence_request_spec.rb [hunk 5]
```diff
@@ -4,7 +4,7 @@
   describe 'validations' do
 
     it { expect(subject).to validate_presence_of :gender }
-    it { expect(subject).to validate_presence_of :date_of_birth }
+    it { expect(subject).to validate_presence_of :detainee_age }
```
Comment (dp6ai) [inline spec/models/defence_request_spec.rb]: this is ridiculous but whatever

**Pull request merged**